File: Pages/home_page.py
class Homepage:
    def __init__(self, page):
        self.page = page

    # Locators
    add_to_cart_buttons_locator = '.button-2.product-box-add-to-cart-button'
    books_menu_locator = 'ul.top-menu.notmobile > li > a[href="/books"]'

    def click_add_to_cart_button(self):
        add_to_cart_button = self.page.locator(self.add_to_cart_buttons_locator).nth(1)
        add_to_cart_button.click()

    # From the webside Top menu, click on books
    def click_books_menu(self):
        self.page.click(self.books_menu_locator)

File: pages/homePage.js
class homePage {
    openUrl() {
        cy.visit(Cypress.env('URL'));
    }

    webLocator = {
        registerTab: '.ico-register',
        loginTab: '.ico-login',
        headerLogo: '.header-logo > a > img',
        booksCatagories: `.top-menu.notmobile [href='/books']`,
        addToCardBtns: '.button-2.product-box-add-to-cart-button',
        shoppingCart: '.cart-label',
        cartQuantity: '#topcartlink .cart-qty',
    }

    goToBooksCatagories() {
        cy.get(this.webLocator.booksCatagories).click({ force: true });
    }

    clickRigesterTab() {
        cy.get(this.webLocator.registerTab).click()
    }

    clickLoginTab() {
        cy.get(this.webLocator.loginTab).click()
    }

    goToShoppingCart() {
        cy.get(this.webLocator.shoppingCart).click({ force: true })
    }

    validateHeader() {
        cy.get(this.webLocator.headerLogo).should('be.visible')
    }

    clickAddToCardBtn() {
        cy.get(this.webLocator.addToCardBtns).eq(2).should('be.visible').click({ force: true });
    }

    addProductInCart() {
        this.clickAddToCardBtn()
    }
  
}
export default new homePage();
